### PR TITLE
add yaml-mime

### PR DESCRIPTION
**Pull request by vwxyzh**
@chenkennt @superyyrrzz @hellosnow @ansyral @qinezh

### CODE REVIEW

File: src/Microsoft.DocAsCode.Common/YamlMime.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+            var line = reader.ReadLine();
+            if (!line.StartsWith("#"))
+            {
+                return null;
+            }
+            return line.TrimStart('#', ' ');
+        }
+
+#if !NetCore
+        public static string ReadMime(string file)
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Common/YamlMime.cs]: check file existence?

**Comment by hellosnow** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Common/YamlMime.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        public const string XRefMap = nameof(XRefMap);
+
+        public static string ReadMime(TextReader reader)
+        {
+            var line = reader.ReadLine();
+            if (!line.StartsWith("#"))
+            {
+                return null;
+            }
+            return line.TrimStart('#', ' ');
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Common/YamlMime.cs]: Commens is serialized in multiple lines In YamlUtility.Serialize(), but only read and return the first line here. Read more lines?

Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Common/YamlMime.cs]: "mime" must be first line

**Comment by superyyrrzz** [comment]
:+1:

File: src/Microsoft.DocAsCode.Common/YamlMime.cs (hunk 2 above)
Comment (ansyral) [inline src/Microsoft.DocAsCode.Common/YamlMime.cs]: if so, need add some comments for the parameter "comments" in YamlUtility that its first line would be used as mime?

---
In reply to: [69884493](https://github.com/dotnet/docfx/pull/476#discussion_r69884493) [](ancestors = 69884493)

Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Common/YamlMime.cs]: yes, first line should be mime, and then other comments.
and there is no difference between "###" and "#" in yaml. just better experience for reader.

**Comment by qinezh** [comment]
:shipit:

**Comment by ansyral** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Common/YamlMime.cs (hunk 1 above)
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Common/YamlMime.cs]: I can't check all cases following, so just let it throw.
```csharp
        // Exceptions:
        //   T:System.UnauthorizedAccessException:
        //     The caller does not have the required permission.
        //
        //   T:System.ArgumentException:
        //     path is a zero-length string, contains only white space, or contains one or more
        //     invalid characters as defined by System.IO.Path.InvalidPathChars.
        //
        //   T:System.ArgumentNullException:
        //     path is null.
        //
        //   T:System.IO.PathTooLongException:
        //     The specified path, file name, or both exceed the system-defined maximum length.
        //     For example, on Windows-based platforms, paths must be less than 248 characters,
        //     and file names must be less than 260 characters.
        //
        //   T:System.IO.DirectoryNotFoundException:
        //     The specified path is invalid, (for example, it is on an unmapped drive).
        //
        //   T:System.IO.FileNotFoundException:
        //     The file specified in path was not found.
        //
        //   T:System.NotSupportedException:
        //     path is in an invalid format.
```

**Pull request merged**